spectral_cube/stokes_spectral_cube.py

```diff
@@ -193,70 +222,96 @@ def _new_cube_with(self, stokes_data=None,
 
     def transform_basis(self, stokes_basis=''):
         """
-        The goal of this function is to transform the stokes basis of the cube to
-        the one specified if possible. In principle one needs at least two related
-        stokes of a given basis for the transformation to be possible. At this moment
-        we are limiting it to the cases where all four stokes parameters in a given
-        basis be available within the cube. The transformations are as follows
-        Linear to Sky
-        I = (XX + YY) / 2
-        Q = (XX - YY) / 2
-        U = (XY + YX) / 2
-        V = 1j(XY - YX) / 2
-
-        Circular to Sky
-        I = (RR + LL) / 2
-        Q = (RL + LR) / 2
-        U = 1j*(RL - LR) / 2
-        V = (RR - LL) / 2
+        Transform the Stokes basis of the cube to the one specified if possible.
+        Operates on the underlying data arrays, not SpectralCube objects.
+        This makes the operation very expensive, so it should be used with care.
         """
-        if self.shape[0] < 4:
+        if len(self._stokes_data) < 4:
             errmsg = "Transformation of a subset of Stokes axes is not yet supported."
-            errmsg_template = "Transformation from {} to {} is not yet supported."
-
             raise NotImplementedError(errmsg)
+
+        # Helper to extract data arrays
+        def get_data(key):
+            return self._stokes_data[key].unmasked_data[:]
+
+        # Helper to create SpectralCube from array
+        def make_cube(array, template_key):
+            return SpectralCube(array, wcs=self._wcs, mask=self._stokes_data[template_key].mask)
+
+        # Transformations
+        if self.stokes_type == "FEED_LINEAR" and stokes_basis == "Sky":
+            XX = get_data('XX')
+            YY = get_data('YY')
+            XY = get_data('XY')
+            YX = get_data('YX')
+            I = (XX + YY) / 2
+            Q = (XX - YY) / 2
+            U = (XY + YX) / 2
+            V = 1j * (XY - YX) / 2
```

Comment:
Following H & B 1996 (https://aas.aanda.org/articles/aas/pdf/1996/07/dst6557.pdf) the sign of V needs to be flipped here, defined as 

```V = -1j * (XY - YX)/2```

Reply:
Done. Check after the current push and mark both as resolved if you agree.